### ISSUE TITLE
Variant of Basic Theme that is Valid HTML5

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#1562]
Would be nice to have `html5validator` without errors on themes inherited from basic (or an alternative to basic). The <tt> and <big> elements are obsolete there.

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/1562
- Originally reported by: Sven Kreiss
- Originally created at: 2014-09-07T20:15:09.478

**Comment by shimizukawa** [comment]
_From Takayuki Shimizukawa on 2014-09-08 14:35:22+00:00_

Thanks for a reporting.
- `<big>` tag was replaced with `<span class="sig-paren">` by 38d58849869a.
- `<tt>` was replaced with `<code>` by https://bitbucket.org/birkenfeld/sphinx/pull-request/269.

These 2 changes will be released by sphinx-1.3.

However, I thinks some other html5 validation errors are exist because sphinx html writer is implemented based upon the `docutils.writers.html4css1` writer.

**Comment by shimizukawa** [comment]
_From Sven Kreiss on 2014-10-24 14:47:33+00:00_

Thanks for those fixes.

There are still many errors that html5validator returns. I am using sphinx==1.3b1 now:

```
> Almost standards mode doctype. Expected “<!DOCTYPE html>”
> error: The “frame” attribute on the “table” element is obsolete. Use CSS instead.
> error: The “rules” attribute on the “table” element is obsolete. Use CSS instead.
> error: The “valign” attribute on the “tbody” element is obsolete. Use CSS instead.
> error: The “frame” attribute on the “table” element is obsolete. Use CSS instead.
> error: Bad value “top” for attribute “rel” on element “link”: The string “top” is not a registered keyword.
> error: The “valign” attribute on the “td” element is obsolete. Use CSS instead.
> error: Element “dl” is missing a required instance of child element “dd”.
> error: The “valign” attribute on the “td” element is obsolete. Use CSS instead.
```

Also see, for example, this report: http://html5.validator.nu/?doc=http%3A%2F%2Fwww.svenkreiss.com%2Fdatabench%2Fv0.3%2Ffrontend.html

**Comment by Suzumizaki** [comment]
Hmm, the issue like this already exists. You can try [sphinx_html5_basic_theme](https://pypi.python.org/pypi/sphinx_html5_basic_theme). The package includes also html5_sphinxdoc theme, but need the patch #1885(see alse #1884).

**Comment by mgifford** [comment]
Surely though the default themes that come with Shinx should be valid html5 though, right? Isn't this issue mostly about cleaning up the defaults rather than alternative themes?

**Comment by tk0miya** [comment]
An experimental HTML5 writer is now introduced at #3464.
Thanks,

**Comment by shibukawa** [comment]
I tested, but there are three errors:

```
Warning: The document role is unnecessary for element body.
From line 36, column 3; to line 36, column 24
</head>↩  <body role="document">↩  ↩↩ 
Error: Start tag span seen in table.
From line 92, column 34; to line 93, column 22
 id="id1">↩<span id="test-table"></span
Fatal Error: Cannot recover after last error. Any further errors will be ignored.
From line 92, column 34; to line 93, column 22
 id="id1">↩<span id="test-table"></span
```

1. Fix theme template
2. Fix table caption with ``<caption>`` tag

**Comment by tk0miya** [comment]
Oh, I jumped to conclusions. Reopened.

**Comment by shibukawa** [comment]
* #3495
* https://sourceforge.net/p/docutils/bugs/312/

**Comment by tk0miya** [comment]
Fixed at #3495. Thank you @shibukawa san!
